### ISSUE TITLE
Return.portfolio subsetting Return xts by 1:ncol(weights)

### ISSUE THREAD

**Issue by raymkchow** [braverock/PerformanceAnalytics#110]
Hi everyone,

Just have a question about behavior of ```Return.portfolio```.
If we have a single name portfolio (only one stock, AMZN.US), with a return xts of many stocks including AMZN.US, I expect that the return of the portfolio will be the same as return of AMZN.US. But as you can see from below, when ``` ncol(price_return) > ncol(weights)```, price_return will be subsetted (by 1:ncol(weights)) , not colnames. Is this behavior by design or a bug?

My version of ```PerformanceAnalytics``` is 1.5.2.

Thanks for the help.

Here is the test code: 
```r
> library(PerformanceAnalytics)
> price_return <- structure(
+   c(NA, 0.00464495151831867, 0.0113853089059701, NA, 
+     0.0044760006643414, 0.0161625013434306, NA, 0.00362131401965859, 
+     0.0145710603829159), 
+   index = structure(c(1514937600, 1515024000, 1515110400), tzone = "UTC", tclass = "Date"), 
+   class = c("xts", "zoo"), .indexCLASS = "Date", tclass = "Date", .indexTZ = "UTC", tzone = "UTC", 
+   .Dim = c(3L, 3L), .Dimnames = list(NULL, c("AAPL.US", "AMZN.US", "GOOG.US"
+   )))
> print(price_return)
               AAPL.US     AMZN.US     GOOG.US
2018-01-03          NA          NA          NA
2018-01-04 0.004644952 0.004476001 0.003621314
2018-01-05 0.011385309 0.016162501 0.014571060
> weights <- xts(matrix(c(1), ncol = ), order.by = as.Date("2018-01-02"))
> colnames(weights) <- c("AMZN.US")
> portfolio_return <- Return.rebalancing(price_return, weights = weights, verbose = FALSE)
Warning messages:
1: In Return.rebalancing(price_return, weights = weights, verbose = FALSE) :
  NA's detected: filling NA's with zeros
2: In Return.rebalancing(price_return, weights = weights, verbose = FALSE) :
  number of assets in beginning_weights is less than number of columns in returns, so subsetting returns.
> print(portfolio_return)
           portfolio.returns
2018-01-03       0.000000000
2018-01-04       0.004644952
2018-01-05       0.011385309
> print(price_return$AMZN.US)
               AMZN.US
2018-01-03          NA
2018-01-04 0.004476001
2018-01-05 0.016162501
```

Ray

**Comment by braverock** [comment]
Thanks for the report.

The code should probably subset by name, if the weights vector is named, and it doesn't appear to.

Patches welcome, or I will try to get to it when I have some time.

**Comment by divyansh997** [comment]
Hi Brian @braverock , I'd like to start working on this issue as a solution for medium test for the project "Multiple plotting engines for PerformanceAnalytics". Any hints on where I should start?(since project proposal date is near, I'd like to complete it ASAP) I was thinking to begin with Return.portfolio.R file in the R folder.

**Comment by braverock** [comment]
This issue is closed by the merging of pull request #116
